ngraph/core/reference/include/ngraph/runtime/reference/random_uniform.hpp

```diff
@@ -0,0 +1,263 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ctime>
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            std::pair<uint32_t, uint32_t> split_high_low(uint64_t value)
```

Comment:
I propose to move implementation to `cpp`, in this case header will have only `evaluate` declaration.

Reply:
Applied this comment in separate PR:  https://github.com/openvinotoolkit/openvino/pull/7012